kola/tests/kubeadm/kubeadm.go

```diff
@@ -124,27 +127,42 @@ etcd:
 )
 
 func init() {
-	for version, params := range testConfig {
-		for _, CNI := range CNIs {
-			// ugly but required to remove the reference between params and the params
-			// actually used by the test.
-			testParams := make(map[string]interface{})
-			for k, v := range params {
-				testParams[k] = v
+	registerTests := func(cgroupv1 bool) {
+		for version, params := range testConfig {
+			if cgroupv1 && !strings.HasPrefix(version, "v1.21") {
+				continue
+			}
+			for _, CNI := range CNIs {
+				if cgroupv1 && !strings.HasPrefix(CNI, "calico") {
+					continue
+				}
+				// ugly but required to remove the reference between params and the params
+				// actually used by the test.
+				testParams := make(map[string]interface{})
+				for k, v := range params {
+					testParams[k] = v
+				}
+				testParams["CNI"] = CNI
+				testParams["Release"] = version
+				cgroupSuffix := ""
+				if cgroupv1 {
+					testParams["cgroupv1"] = true
+					cgroupSuffix = ".cgroupv1"
+				}
+
+				register.Register(&register.Test{
+					Name:             fmt.Sprintf("kubeadm.%s.%s%s.base", version, CNI, cgroupSuffix),
+					Distros:          []string{"cl"},
+					ExcludePlatforms: []string{"esx"},
+					Run: func(c cluster.TestCluster) {
+						kubeadmBaseTest(c, testParams)
+					},
```

Comment:
As in previous commit, I think we need a minimum version ?

Reply:
I've added a minversion as well.